codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -139,10 +139,17 @@ public void channelActive(ChannelHandlerContext ctx) throws Exception { }
 
         public void channelInactive(ChannelHandlerContext ctx) throws Exception {
             try {
-                ChannelFuture future = ctx.newSucceededFuture();
-                final Collection<Http2Stream> streams = connection().activeStreams();
-                for (Http2Stream s : streams.toArray(new Http2Stream[streams.size()])) {
-                    closeStream(s, future);
+                final Http2Connection connection = connection();
+                // Check if there are streams to avoid the overhead of creating the ChannelFuture.
+                if (connection.numActiveStreams() != 0) {
```

Comment:
@Scottmitch why is this needed? what's `the overhead of creating the ChannelFuture.`?